rta/src/main/java/com/gluonhq/richtextarea/viewmodel/RichTextAreaViewModel.java

```diff
@@ -755,7 +755,8 @@ private void updateProperties() {
 
     private Document getCurrentDocument(Selection selection) {
         // text and indices should be based on the exportable text
-        int caret = getTextBuffer().getText(0, getCaretPosition()).length();
+        int caretPosition = getCaretPosition() < 0 ? getTextLength() : getCaretPosition();
+        int caret = getTextBuffer().getText(0, caretPosition).length();
```

Comment:
If the caret position is less than 0 then we auto-assume that it will be at the end of the text? how does this assumption help? I can think of a scenario where if a document has a text then RTA should ideally show the caret at the end of the text.

Reply:
This method is `getCurrentDocument`, and we need to get the document: the selected text, its decoration model, and finally the caret, which is not that much relevant. 

`getCaretPosition() < 0` means that caret is -1, or not defined. In that case, we simply put it at the end of the document. 

This change was needed to prevent calling `getTextBuffer().getText(0, -1)` in such case.